lib/util.js

```diff
@@ -748,27 +748,74 @@ var util = {
   /**
    * @api private
    */
-  addPromisesToRequests: function addPromisesToRequests(constructor, PromiseDependency) {
-    PromiseDependency = PromiseDependency || null;
-    if (!PromiseDependency && typeof Promise !== 'undefined') {
-      PromiseDependency = Promise;
-    }
-    if (typeof PromiseDependency !== 'function') {
-      delete constructor.prototype.promise;
-      return;
-    }
-    constructor.prototype.promise = function promise() {
-      var self = this;
-      return new PromiseDependency(function(resolve, reject) {
-        self.on('complete', function(resp) {
-          if (resp.error) {
-            reject(resp.error);
-          } else {
-            resolve(resp.data);
-          }
+  addPromises: function addPromises(constructors, PromiseDependency) {
+    if (!AWS) AWS = require('./core');
+    if (!AWS.config) require('./config');
+    if (PromiseDependency === undefined) PromiseDependency = AWS.config.getPromisesDependency();
+    if (typeof PromiseDependency !== 'function') var deletePromise = true;
+    if (!Array.isArray(constructors)) constructors = [constructors];
+
+    var promisifyMethod = function(methodName) {
+      return function promise() {
+        var self = this;
+        return new PromiseDependency(function(resolve, reject) {
+          self[methodName](function(err, data) {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(data);
+            }
+          });
         });
-        self.runTo();
-      });
+      };
+    };
+
+    for (var ind = 0; ind < constructors.length; ind++) {
+      var constructor = constructors[ind];
+      switch (constructor.name) {
```

Comment:
The reason I originally checked `constructor.name` is because I hadn't originally planned on requiring `core.js`, so I didn't have access to the actual constructors for an equality check before I added that require statement. I may be refactoring the require out anyway (as discussed above). I'll look into adding the static method for each class.